src/nomad_simulations/schema_packages/properties/thermodynamics.py

```diff
@@ -0,0 +1,289 @@
+    Physical property section describing the enthalpy (i.e. internal_energy + pressure * volume.) of a (sub)system.
+    """
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class Entropy(PhysicalProperty):
+    """
+    Physical property section describing the entropy of a (sub)system.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        unit='joule / kelvin',
+        description="""
+        The value of the entropy.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class GibbsFreeEnergy(BaseEnergy):
+    """
+    Physical property section describing the Gibbs free energy (i.e., enthalpy - temperature * entropy) of a (sub)system.
+    """
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class HelmholtzFreeEnergy(BaseEnergy):
+    """
+    Physical property section describing the Gibbs free energy (i.e., internal_energy - temperature * entropy) of a (sub)system.
+    """
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class ChemicalPotential(BaseEnergy):
+    """
+    Free energy cost of adding or extracting a particle from a thermodynamic system.
+    """
+
+    # ! implement `iri` and `rank` as part of `m_def = Section()`
+
+    iri = 'http://fairmat-nfdi.eu/taxonomy/ChemicalPotential'
+
+    def __init__(
+        self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        self.rank = []
+        self.name = self.m_def.name
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class HeatCapacityCV(PhysicalProperty):
+    """
+    Physical property section describing the heat capacity at constant volume for a (sub)system.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        unit='joule / kelvin',
+        description="""
+        The value of the heat capacity.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class HeatCapacityCP(PhysicalProperty):
+    """
+    Physical property section describing the heat capacity at constant volume for a (sub)system.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        unit='joule / kelvin',
+        description="""
+        The value of the heat capacity.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+################################
+# other thermodynamic properties
+################################
+
+
+class Virial(BaseEnergy):
```

Comment:
What is this property? Is it: https://en.wikipedia.org/wiki/Virial_expansion?

Reply:
Ah, very cool. I think a short but complete description is enough. 

Two points here:

1. If this is a tensor of order 2, then you should define `self.rank`. This should be a definition, but for now I was keeping rank defined in the initialization `__init__` of the class (you can check out hopping_matrix.py for an example). I guess that in this case is as simple as `self.rank = [3, 3]`
2. The traces (or well, 1/3 of the trace) could perfectly be other properties. This is something we discussed in the taxonomy tf, but also it makes sense being that `rank` needs to be a definition of the property. In the trace case, `self.rank = []`. I also wonder if the trace has a specific name or it should be just the `TraceVirial`

Reply:
Any idea what is wrong with the test: https://github.com/nomad-coe/nomad-simulations/actions/runs/9777748430/job/26993079402?pr=91

I just did:
```python
    def __init__(
        self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs
    ) -> None:
        super().__init__(m_def, m_context, **kwargs)
        self.rank = [3, 3]
        self.name = self.m_def.name
```

Reply:
What about changing the import of `BaseEnergy` to be to `energies` instead of properties?

```python
from nomad_simulations.schema_packages.energies import BaseEnergy
```

It might be a circular import issue.

You can also run the action locally:
```sh
python -m mypy --ignore-missing-imports --follow-imports=silent --no-strict-optional src/nomad_simulations/schema_packages tests
```
before pushing to test if it passes or not.

Reply:
As @JosePizarro3 stated, or you add `BaseEnergy` to `properties.__init__.py` to expose it.

Reply:
>  src/nomad_simulations/schema_packages/properties/thermodynamics.py:196: error: Need type annotation for "rank" (hint: "rank: List[<type>] = ...")  [var-annotated]

The 2nd eror is pretty straight-forward: you should specify with which type you will be populating the list with.

Reply:
> The 2nd eror is pretty straight-forward: you should specify with which type you will be populating the list with.

@JFRudzinski try first the import thing. I didn't need to define the type of rank so it might be a problem with imports.

Reply:
Yes the import works thanks, I forgot that I had removed BaseEnergy from init